code/main/task/eventProcessor.brs

```diff
@@ -247,23 +195,107 @@ function _adb_EventProcessor(task as object) as object
         _sendResponseEvent: function(event as object) as void
             _adb_logInfo("EventProcessor::_sendResponseEvent() - Sending response event: (" + chr(10) + FormatJson(event) + chr(10) + ")")
 
-            if m._task = invalid
-                _adb_logError("EventProcessor::_sendResponseEvent() - Cannot send response event, task node instance is invalid.")
-                return
-            end if
-
             if _adb_isResponseEvent(event)
-                m._task[m._CONSTANTS.TASK.RESPONSE_EVENT] = event
+                ''' for public API response events, we need to set the response event in task node
+                m._dispatchResponseEventToTask(event)
+
+                ''' for modules response events, we need to call handleResponseEvent
+                m._dispatchResponseEventToRegisteredModules(m._modulesRegisteredForResponseEvents, event)
             else
                 _adb_logError("EventProcessor::_sendResponseEvent() - Cannot send response event, invalid event:(" + chr(10) + FormatJson(event) + chr(10) + ")")
             end if
         end function,
 
+        _dispatchResponseEventToTask: function(event as object) as void
+            if m._task = invalid
+                _adb_logError("EventProcessor::_dispatchResponseEventToTask() - Cannot send response event, task node instance is invalid.")
+                return
+            end if
+
+            m._task[m._CONSTANTS.TASK.RESPONSE_EVENT] = event
+        end function,
+
+        _dispatchResponseEventToRegisteredModules: function(registeredModules as object, event as object) as void
+            _adb_logVerbose("EventProcessor::_dispatchResponseEventToRegisteredModules() - Received response event:(" + chr(10) + FormatJson(event) + chr(10) + ")")
+
+            ''' notify response events all the modules
+            for each module in registeredModules
+                ''' registered modules should have the method processResponseEvent
+                module.processResponseEvent(event)
+            end for
+
+        end function,
+
         _sendResponseEvents: function(responseEvents as dynamic) as void
+            if _adb_isEmptyOrInvalidArray(responseEvents)
+                return
+            end if
+
             for each event in responseEvents
                 m._sendResponseEvent(event)
             end for
         end function,
+
+        ''' TODO cleanup (clean up Media public API data and MediaModule)
+        ''' Validates the event data for createMediaSession request event
+        _isValidEventDataForSessionStartRequest: function(data as object) as boolean
```

Comment:
I prefer exposing these two helpers from the media module rather than defining them in this class. You can include this change as part of your media cleanup..

Reply:
YesI have some cleanups regarding similar things. I will add it to that.

Reply:
https://github.com/adobe/aepsdk-roku/issues/175